src/cursor.rs

```diff
@@ -57,12 +57,12 @@ pub trait Cursor<'txn> {
     /// For databases with duplicate data items (`DatabaseFlags::DUP_SORT`), the
     /// duplicate data items of each key will be returned before moving on to
     /// the next key.
-    fn iter_from<K>(&mut self, key: K) -> Result<Iter<'txn>> where K: AsRef<[u8]> {
+    fn iter_from<K>(&mut self, key: K) -> Box<Iterator<Item=Result<(&'txn [u8], &'txn [u8])>>> where K: AsRef<[u8]> {
```

Comment:
The way to avoid the `Box` is easy enough -- push the branch down into the underlying `Iter` struct.  That is, make it:
```
/// An iterator over the values in an LMDB database.
pub enum Iter<'txn> {
    Err(Result::E),
    Ok(...),
}
```
where the bits that are currently in `Iter` are in the `Ok` branch as well.

Reply:
> The way to avoid the `Box` is easy enough -- push the branch down into the underlying `Iter` struct.

Ok, I've done this in a558fab.